src/components/search/searchPageSearchForm/SearchPageSearchForm.tsx

```diff
@@ -40,11 +43,17 @@ function SearchPageSearchForm() {
 
   const doSearch = (q?: string) => {
     const nextQuery = q ? { q } : null;
+    const params = getURLSearchParamsFromAsPath(router.asPath);
 
     queryPersister.persistQuery(nextQuery);
-    router.push({ pathname: router.pathname, query: nextQuery }, undefined, {
-      shallow: true,
-    });
+    params.set("q", nextQuery.q);
```

Comment:
Here you are accidentally introducing a bug by not persisting all parameters. If the user were to select a place from the map and click back, they would be taken into the list view instead of the map view.

Reply:
I had a feeling I would mess up something with the query persister 😄 fixing this as well.